src/core/stompest/protocol/frame.py

```diff
@@ -106,7 +111,7 @@ def version(self, value):
         self._version = StompSpec.version(value)
 
     def _encode(self, text):
-        return StompSpec.CODECS[self.version].encode(text)[0]
```

Comment:
Please factor out the distinction of cases s.t. no redundant code remains, e.g. by appending an `if` statement which decodes the result.